### PR TITLE
Various fixes, address todos, flake8

### PR DESCRIPTION
**Pull request by scanon**

### CODE REVIEW

File: JobRunner/DockerRunner.py [hunk 1]
```diff
@@ -21,30 +18,41 @@ def __init__(self, logger=None):
...
 
     def _sort_logs(self, sout, serr):
         """
         This is an internal function to sort and interlace output for NJS.
         This is not fully implemented yet and sould be rethought.
         """
-        # TODO: Fix sorting
 
-        lines = []
+        tslines = dict()
```
Comment (bio-boris) [inline JobRunner/DockerRunner.py]: What does tslines stand for?

Comment (scanon) [reply JobRunner/DockerRunner.py]: time stamped lines.  We could make it longer.

File: JobRunner/DockerRunner.py [hunk 2]
```diff
@@ -21,30 +18,41 @@ def __init__(self, logger=None):
...
+                    (ts, txt) = line.split(maxsplit=1)
+                    if ts not in tslines:
+                        tslines[ts] = []
+                    tslines[ts].append({'line': txt, 'is_error': ierr})
+        nlines = []
+        for ts in sorted(tslines.keys()):
+            nlines.extend(tslines[ts])
+        return nlines
 
     def _shepherd(self, c, job_id, subjob, queues):
```
Comment (bio-boris) [inline JobRunner/DockerRunner.py]: Could we come up with a better name for this function? Maybe something like log_shepherd or container_log_manager ?

Comment (bio-boris) [reply JobRunner/DockerRunner.py]: It looks like subjob is not used.

File: JobRunner/MethodRunner.py [hunk 3]
```diff
@@ -140,11 +157,17 @@ def run(self, config, module_info, params, job_id, fin_q=None,
...
+
         env = {
             'SDK_CALLBACK_URL': callback
         }
-        # TODO: Add secure params
+
+        # Add secure params
+        if module_info.get('secure_config_params') is not None:
+            for p in module_info['secure_config_params']:
+                k = 'KBASE_SECURE_CONFIG_PARAM_%s' % (p['param_name'])
```
Comment (bio-boris) [inline JobRunner/MethodRunner.py]: FYI you can use fstrings instead of these substitutions

**Comment by bio-boris** [comment]
* Changes look good.
* Tests look good. 
* More descriptive names would be helpful to increase understanding of what is going on in some of these methods

**Pull request merged**